src/vec.rs

```diff
@@ -534,6 +535,59 @@ where
     }
 }
 
+impl<ID, Element> IdentifiedVec<ID, Element>
+where
+    ID: Eq + Hash + Clone + Debug,
+    Element: Eq + Hash + Clone + Debug,
```

Comment:
Unfortunately not. It gives the following error:

`the trait bound `ID: Hash` is not satisfied
the trait `Hash` is not implemented for `ID`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Fviktorjansson%2FDeveloper%2FRust%2Fidentified_vec%2Fsrc%2Fvec.rs)
vec.rs(182, 14): required by a bound in `IdentifiedVec`
vec.rs(540, 11): consider further restricting this bound: ` + std::hash::Hash`
the trait bound `ID: Clone` is not satisfied
the trait `Clone` is not implemented for `ID`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B3%5D?3#file%3A%2F%2F%2FUsers%2Fviktorjansson%2FDeveloper%2FRust%2Fidentified_vec%2Fsrc%2Fvec.rs)
vec.rs(182, 21): required by a bound in `IdentifiedVec`
vec.rs(540, 11): consider further restricting this bound: ` + std::clone::Clone`
`ID` doesn't implement `Debug`
`ID` cannot be formatted using `{:?}` because it doesn't implement `Debug`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B6%5D?6#file%3A%2F%2F%2FUsers%2Fviktorjansson%2FDeveloper%2FRust%2Fidentified_vec%2Fsrc%2Fvec.rs)
vec.rs(182, 29): required by a bound in `IdentifiedVec`
vec.rs(540, 11): consider further restricting this bound: ` + std::fmt::Debug`

Reply:
I meant to should keep the constraints on ID as usual, aand add only Eq to Element

Reply:
i.e.:

```rust
impl<ID, Element> IdentifiedVec<ID, Element>
where
    ID: Eq + Hash + Clone + Debug,
    Element: Eq
```

Reply:
Ah, my bad. `Element` needs `Debug` too in order to do this: `return Err(Error::ElementWithSameValueFound(format!("{:#?}", value)));` 

So it would be:
```
impl<ID, Element> IdentifiedVec<ID, Element>
where
    ID: Eq + Hash + Clone + Debug,
    Element: Eq  
    {
```